src/js/services/optimizely.test.js

```diff
@@ -0,0 +1,265 @@
+        };
+        window.dispatchEvent(userDataReady);
+      }, 100);
+
+      await expect(client.getUserId()).resolves.toBe(
+        '00000000-0000-0000-0000-000000000000',
+      );
+    });
+  });
+
+  // Function used in TrackExperimentViewed
+  describe('isExperimentAlreadyViewed()', () => {
+    beforeEach(() => {
+      // ensure local storage clear
+      window.localStorage.clear();
+    });
+    it('experiment has not been viewed before', () => {
+      storeExperimentParticipation(orgId, experimentKey + '456', variationName);
+      expect(isExperimentAlreadyViewed(orgId, experimentKey)).toBe(false);
+    });
+    it('experiment has been viewed before', () => {
+      storeExperimentParticipation(orgId, experimentKey, variationName);
+      expect(isExperimentAlreadyViewed(orgId, experimentKey)).toBe(true);
+    });
+  });
+
+  //Function used in getVariationName
+  describe('trackExperimentViewed()', () => {
+    beforeEach(() => {
+      global.AnalyticsClient = AnalyticsClient;
+      window.localStorage.clear();
+      jest.clearAllMocks();
+    });
+
+    afterEach(() => {
+      delete global.AnalyticsClient;
+      jest.resetAllMocks();
+    });
+
+    it('expect trackExperimentViewed to have bene viewed', () => {
+      storeExperimentParticipation(orgId, experimentKey, variationName);
+      expect(isExperimentAlreadyViewed(orgId, experimentKey)).toBe(true);
+
+      const spy = jest.spyOn(window.AnalyticsClient, 'trackAction');
+      trackExperimentViewed(
+        orgId,
+        experimentKey,
+        ['container item'],
+        '5',
+        variationName,
+        '5',
+        '5',
+      );
+      expect(spy).not.toHaveBeenCalled();
+    });
+
+    it('expect trackExperimentViewed to have not been viewed', () => {
+      const spy = jest.spyOn(window.AnalyticsClient, 'trackAction');
+      trackExperimentViewed(
+        orgId,
+        experimentKey,
+        ['container item'],
+        '5',
+        variationName,
+        '5',
+        '5',
+      );
+      expect(spy).toHaveBeenCalled();
+    });
+  });
+
+  describe('getVariationName()', () => {
+    afterEach(() => {
+      jest.clearAllMocks();
+      jest.resetAllMocks();
+    });
+
+    it('test forceAll is true return treatment', async () => {
+      forceAll.mockImplementation(() => true);
+      await expect(client.getVariationName(options)).resolves.toBe('treatment');
+      expect(forceAll.mock.calls.length).toBe(1);
+    });
+
+    it('test forceAll is false options null', async () => {
+      const errorObject = { error: 'Missing required options' };
+      forceAll.mockImplementation(() => false);
+      await expect(client.getVariationName(null)).rejects.toEqual(errorObject);
+    });
+
+    it('test forceAll is false valid options no cookie', async () => {
+      forceAll.mockImplementation(() => false);
+      await expect(client.getVariationName(options)).resolves.toBe(null);
+      expect(forceAll.mock.calls.length).toBe(1);
+    });
```

Comment:
I'm not sure I understand what this is testing?

Reply:
Specifically I was testing if there is no orgId because there is no cookie saved
      if (!orgId) {
        return resolve(null);
      }
 It was a branch test, I can make the test message more clear.